frontend/models/product.ts

```diff
@@ -25,6 +25,15 @@ export async function findProduct(shop: ShopCredentials, handle: string) {
       return null;
    }
    const variants = getVariants(response.product);
+   const activeVariants = variants.filter(
+      (variant) =>
+         !variant.disableWhenOOS ||
+         (variant.quantityAvailable && variant.quantityAvailable > 0)
+   );
+
+   const images = getImages(response.product.images, variants);
```

Comment:
Do we want to only get images for active variants here?

```suggestion
   const images = getImages(response.product.images, activeVariants);
```